src/Pact/Types/Type.hs

```diff
@@ -382,7 +384,7 @@ isVarTy _ = False
 -- 'a unifiesWith b' means 'a' "can represent/contains" 'b'.
 -- In use, 'a' is the spec, and 'b' is the value type being unified with 'a'.
 -- First argument is 'Eq1' predicate.
-unifiesWith :: (n -> n -> Bool) -> Type n -> Type n -> Bool
+unifiesWith :: Eq1 Type => (n -> n -> Bool) -> Type n -> Type n -> Bool
```

Comment:
`liftEq` wants it. It's because the TH generated instances from the bottom of the module are not in scope at this point with GHC >=9. I could try to move this function to the bottom of the module.